### PR TITLE
yoav - motion planner

### PR DESCRIPTION
**Pull request by greenblitz4590**
Before issuing a pull request, please see the contributing page.

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java [hunk 1]
```diff
@@ -35,16 +37,15 @@ public static Robot getInstance() {
     private Wrist wrist;
 
     private void initSubsystems(HardwareMap hardwareMap) {
-//        if (currentState == null) {
```
Comment (GilStein1) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java]: remove extra tab

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/StateMotionPlanner.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package org.firstinspires.ftc.teamcode;
+
+public class StateMotionPlanner {
+
+    private int robotStateIndex;
+
+    public StateMotionPlanner(RobotState startingState) {
```
Comment (GilStein1) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/StateMotionPlanner.java]: protected

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/StateMotionPlanner.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        this.robotStateIndex = getStateIndex(startingState);
+    }
+
+    protected RobotState getLeftState() {
```
Comment (GilStein1) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/StateMotionPlanner.java:13]: find a solution to the magic numbers

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/robotstatetesters/RazMotionPlannerTest.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    @Override
+    public void execute() {
+        telemetry.addData("index", Robot.getInstance().getStateMotionPlanner().getRobotStateIndex());
+        telemetry.addData("state", Robot.getInstance().getCurrentState());
+    }
+
+    @Override
+    public void configureBindings() {
+        Bindings.razMotionPlannerTest(gamepad2);
+    }
```
Comment (GilStein1) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/robotstatetesters/RazMotionPlannerTest.java:26]: space at the end of the class

**Review by GilStein1 — COMMENTED** [review]
code

**Review by GilStein1 — DISMISSED** [review]
very good, very proud

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java [hunk 5]
```diff
@@ -1,7 +1,9 @@
 package org.firstinspires.ftc.teamcode;
 
+import com.arcrobotics.ftclib.command.Command;
 import com.arcrobotics.ftclib.command.SequentialCommandGroup;
 import com.qualcomm.robotcore.hardware.HardwareMap;
+
```
Comment (YoniKiriaty) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java:6]: why? remove space

Comment (YHerm) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java:6]: It’s for separating between team code import to library import, I think formater do it automatically

**Review by YoniKiriaty — APPROVED** [review]

**Pull request merged**